src/Connection.php

```diff
@@ -345,4 +345,13 @@ private function fireEvent($event, array $args)
 		}
 	}
 
+
+	/**
+	 * @return SqlProcessor
+	 */
+	public function getSqlPreprocessor()
+	{
+		return $this->sqlPreprocessor;
+	}
+
```

Comment:
yes, it won't work correctly, ie. reconnect will recreate the instance without custom modifiers.